devito/propagator.py

```diff
@@ -4,20 +4,48 @@
 from sympy import symbols, IndexedBase, Indexed
 from function_manager import FunctionDescriptor
 from collections import Iterable
+from at_controller import get_at_block_size, get_optimal_block_size
 
 
 class Propagator(object):
-    def __init__(self, name, nt, shape, spc_border=0, forward=True, time_order=0, openmp=False, profile=False, cache_blocking=False, block_size=5):
+    def __init__(self, name, nt, shape, spc_border=0, forward=True, time_order=0, openmp=False, profile=False,
+                 cache_blocking=False, block_size=None, auto_tune=False):
         self.t = symbols("t")
+
         self.cache_blocking = cache_blocking
-        if(isinstance(block_size, Iterable)):
-            if(len(block_size) == len(shape)):
-                self.block_sizes = block_size
+        self.auto_tune = auto_tune
+
+        self.spc_order = spc_border * 2  # setting space order
+        self.time_order = time_order
+
+        if self.cache_blocking:
```

Comment:
No amount of documentation will help a bad interface. The behaviour should be as expected. If I am not setting auto_tune, picking up values from an old auto-tune run will only confuse the user at best. At worst, it could lead to weird issues that take hours to debug. All because devito found some files lying around from a different run on potentially different hardware (e.g. shared filesystem on a heterogeneous cluster)?